packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -31,7 +31,7 @@ export function useFetch<
   ResT = void,
   ErrorT = FetchError,
   ReqT extends NitroFetchRequest = NitroFetchRequest,
-  Method extends AvailableRouterMethod<ReqT> = 'get' extends AvailableRouterMethod<ReqT> ? 'get' : AvailableRouterMethod<ReqT>,
+  Method extends AvailableRouterMethod<ReqT> = AvailableRouterMethod<ReqT>,
```

Comment:
What about this?

```suggestion
  Method extends AvailableRouterMethod<ReqT> = ResT extends void ? 'get' extends AvailableRouterMethod<ReqT> ? 'get' : AvailableRouterMethod<ReqT> : AvailableRouterMethod<ReqT>,
```

Reply:
I also find the reason of broken type when passing uppercase method, maybe its a nitro type error in `fetch.ts`.
Another PR in [nitro](https://github.com/unjs/nitro/pull/1247)

Reply:
Nice work - thanks! I think we can go ahead and merge this PR (as it fixes _one_ issue without introducing any regressions) and then update test once nitro PR is merged..